docs: disqualify HDD-backed EBS volumes (sc1/st1) for Zeebe (#9078)
camunda/camunda-docs: docs/self-managed/reference-architecture/kubernetes.md

- **Persistent volumes**
  - 1,000 IOPS
  - 32 GiB

>   - Use **SSD-backed** volumes only. HDD-backed volume types (such as AWS EBS `sc1` and `st1`) and burstable disk types cannot meet Zeebe's Raft flush latency requirements and are not supported.

Reviewer: The hard "...and burstable disk types ... are not supported" conflicts with the `gp2` alternative recommended in the EKS section below (around line 308): `gp2` is IOPS-burstable and only reaches ">1,000 IOPS" at 34 GiB through burst credits. Consider scoping the unsupported statement to HDD-backed types, e.g.:

> Use SSD-backed volumes only. HDD-backed volume types (such as AWS EBS `sc1` and `st1`) cannot meet Zeebe's Raft flush latency requirements and are not supported.

Also worth making this rule authoritative beyond the K8s ref-arch (see review summary: `supported-environments.md` + `manual.md`).

(Applies to all four versions: docs/ L190, 8.9 L190, 8.8 L184, 8.7 L139.) Nit: the style guide reserves bold for UI elements, not emphasis — `SSD-backed` could be plain text.

Reviewer: Resolved in 6639342 — the hard "not supported" is now scoped to HDD-backed types and the burst caveat is a provider-neutral note, so the recommended `gp2` fallback is no longer contradicted.

>   - Use SSD-backed volumes. HDD-backed volume types cannot meet Zeebe's Raft flush latency requirements and are not supported. Avoid burstable volume types unless sized to sustain the required IOPS without relying on burst credits.

Reviewer 2: ```suggestion
  - SSD-backed volumes only. HDD-backed volumes are not supported.
  - Avoid burstable volume types unless they can sustain 1,000+ IOPS continuously without relying on burst credits.

:::note
HDD-backed volumes cannot meet Zeebe's Raft protocol disk flush requirements, which demand consistent single-digit-millisecond write latency.
:::
```

Reviewer: Applied in 6639342 ✅ — also backported to 8.9/8.8/8.7. Thanks @zaycodes!

- Volume type: SSD
  - 1,000–3,000 IOPS per volume
  - Throughput of 1,000 MB/s per volume

> - **Unsupported volume types**: HDD-backed volumes cannot meet Zeebe's Raft flush latency requirements. Choose an SSD-backed storage class.

Reviewer 2: ```suggestion
- Unsupported volume types: HDD-backed volumes are not supported.
```

Reviewer 2: All other bullets in this block use plain Label: value format (`Volume type: SSD, Volume alternative: ...`). Bolding only this label is inconsistent. The second sentence ("Choose an SSD-backed storage class") is already covered by Volume type: SSD above it.

Reviewer: Applied in 6639342 ✅ — also backported to 8.9/8.8/8.7. Thanks @zaycodes!

  - Only if `gp3` isn't available
  - IOPS performance [varies based on volume size](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/general-purpose.html#gp2-performance)
  - Minimum 34 GiB for >1,000 IOPS

> - **Unsupported volume types**: HDD-backed `sc1` (Cold HDD) and `st1` (Throughput HDD). These volume types cannot meet the synchronous disk flush requirements of Zeebe's Raft consensus protocol. Additionally, `sc1` and `st1` use burst credits for throughput — once exhausted, write latency spikes to hundreds of milliseconds, causing persistent Raft append timeouts and leader instability.

Reviewer: Accurate and useful callout, but it's currently EKS-only. To avoid AWS-only guidance, please either mirror an equivalent "unsupported HDD volume types" note to the AKS (Standard HDD) and GKE (Standard PD / HDD) sections, or rely solely on the generic note above (which already covers all providers).

Nit: the siblings here are plain (`Instance type:`, `Volume type:`, `Volume alternative:`), so a plain `Unsupported volume types:` would match (style guide: bold is for UI elements, not emphasis).

(Same added bullet exists in all four versions: docs/ L311, 8.9 L311, 8.8 L301, 8.7 L253.)

Reviewer: Addressed in 6639342 — per-provider HDD callouts now cover OpenShift, EKS, AKS, and GKE alongside the generic note, across next/8.9/8.8/8.7.

  - Only if `gp3` isn't available
  - IOPS performance [varies based on volume size](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/general-purpose.html#gp2-performance)
  - Minimum 34 GiB for >1,000 IOPS

> - **Unsupported volume types**: HDD-backed `sc1` (Cold HDD) and `st1` (Throughput HDD). These volume types cannot meet the synchronous disk flush requirements of Zeebe's Raft consensus protocol. Additionally, `sc1` and `st1` use burst credits for throughput — once exhausted, write latency spikes to hundreds of milliseconds, causing persistent Raft append timeouts and leader instability.

Reviewer 2: ```suggestion
- Unsupported volume types: `sc1` (Cold HDD) and `st1` (Throughput HDD) are not supported.

:::caution
`sc1` and `st1` cannot meet Zeebe's Raft protocol disk flush requirements. They use burst credits for throughput, and once credits are exhausted, write latency spikes to hundreds of milliseconds, causing persistent Raft append timeouts and leader instability.
:::
```

Reviewer 2: Three sentences of dense technical explanation in a bullet list break scannability. A :::caution callout is appropriate here given this was a root cause of a production incident.

Reviewer: Applied in 6639342 ✅ — also backported to 8.9/8.8/8.7. Thanks @zaycodes!

- Volume alternative: Premium SSD
  - IOPS performance [varies based on volume size](https://learn.microsoft.com/en-us/azure/virtual-machines/disks-types#premium-ssds)
  - Minimum 256 GiB (P15) for > 1,000 IOPS

> - **Unsupported volume types**: Standard HDD cannot meet Zeebe's Raft flush latency requirements. Avoid Standard SSD unless sized to sustain the required IOPS without relying on bursting.

Reviewer 2: ```suggestion
- Unsupported volume types: Standard HDD is not supported.
- Volume alternative caveat: Standard SSD is supported if sized to sustain 1,000+ IOPS continuously without relying on bursting.
```

Reviewer: Applied in 6639342 ✅ — also backported to 8.9/8.8/8.7. Thanks @zaycodes!

- Volume type: Performance (SSD) persistent disks
  - IOPS performance [varies based on volume size](https://cloud.google.com/compute/docs/disks/performance#performance_factors)
  - Minimum 34 GiB for > 1,000 IOPS

> - **Unsupported volume types**: Standard persistent disks (`pd-standard`, HDD-backed) cannot meet Zeebe's Raft flush latency requirements. Use SSD-backed (`pd-ssd`) volumes.

Reviewer 2: ```suggestion
- Unsupported volume types: Standard persistent disks (`pd-standard`, HDD-backed) cannot meet Zeebe's Raft flush latency requirements. Use SSD-backed (`pd-ssd`) volumes.
```

Reviewer: Applied in 6639342 ✅ — also backported to 8.9/8.8/8.7. Thanks @zaycodes!
